### PR TITLE
Add optional defaultValue for getBlob* functions

### PR DESCRIPTION
**Pull request by francescopersico**

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Please include a Pivotal story at the beginning of the PR title (see below).</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>Please include a description of your PR changes.</td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c3d6ebdc69fad1fe626f7cf06597b3b169828b92
</p>

**Comment by francescopersico** [comment]
@cloudify please review.

File: src/utils/azure_storage.ts [hunk 1]
```diff
@@ -161,17 +178,20 @@ export function getBlobAsText(
...
  * @param blobName        blob file name
+ * @param defaultValue    if you want a default value insted of BlobNotFound error
  */
 export async function getBlobAsObject<A, O, I>(
   type: t.Type<A, O, I>,
   blobService: azureStorage.BlobService,
   containerName: string,
-  blobName: string
+  blobName: string,
+  defaultValue?: O
```
Comment (cloudify) [inline src/utils/azure_storage.ts]: I don't think it makes sense to handle the default here, it should be up to the caller to decide, this function should have the minimal responsibilities - beside you can just call `getOrElse` on the `Either`

File: src/utils/azure_storage.ts [hunk 2]
```diff
@@ -161,17 +178,20 @@ export function getBlobAsText(
...
 ): Promise<Either<Error, A>> {
```
Comment (cloudify) [inline src/utils/azure_storage.ts]: what we could do is threat the not found error as a special case and return: `Promise<Either<Error, Option<A>>>` - not found errors will be mapped to `Right<None>` (we do that in other places too like when we query cosmos or redis)

Comment (francescopersico) [reply src/utils/azure_storage.ts]: Implemented in https://github.com/teamdigitale/io-functions-commons/pull/22

**Comment by cloudify** [comment]
I'll close this

File: src/utils/azure_storage.ts [hunk 3]
```diff
@@ -138,15 +144,26 @@ export function upsertBlobFromObject<T>(
...
+  defaultValue?: string
 ): Promise<Either<Error, Option<string>>> {
   return new Promise(resolve => {
     blobService.getBlobToText(containerName, blobName, (err, result, __) => {
       if (err) {
+        // tslint:disable-next-line: no-any
+        const errorAsAny = err as StorageError;
+        if (
+          defaultValue !== undefined &&
+          errorAsAny.code !== undefined &&
```
Comment (gunzip) [inline src/utils/azure_storage.ts]: isn't this check redundant ?

**Pull request closed without merging**